### PR TITLE
added mongodb authentication

### PR DESCRIPTION
**Pull request by SamuelTJackson**
This PR adds MongodDB authentication support

### CODE REVIEW

File: doc/installationguide.md [hunk 1]
```diff
@@ -254,6 +254,9 @@ overrides.
 | IOTA_MONGO_PORT           | `mongodb.port`                  |
 | IOTA_MONGO_DB             | `mongodb.db`                    |
 | IOTA_MONGO_REPLICASET     | `mongodb.replicaSet`            |
+| IOTA_MONGO_USER 	    | `mongodb.user`		      |
```
Comment (fgalan) [inline doc/installationguide.md]: Myabe you could add some information about the new parameters in the `mongodb` section above (around line 136 in this same file).

File: doc/installationguide.md [hunk 2]
```diff
@@ -254,6 +254,9 @@ overrides.
...
+| IOTA_MONGO_PASSWORD	    | `mongodb.password`	      |
+| IOTA_MONGO_AUTH_DB 	    | `mongodb.authDB`		      |
```
Comment (fgalan) [inline doc/installationguide.md]:
```suggestion
| IOTA_MONGO_AUTH_DB 	    | `mongodb.authDb`		      |
```

Not fully sure... we use camelCase as convention, but in the case of acronyms (and DB is an acronym here) I'm not sure how to apply it. But looking other cases (e.g. `clientId`) I think is better this way :)

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -3,3 +3,4 @@ Add NGSIv2 metadata support to device provisioned attributes
 Fix: Error message when sending measures with unknown/undefined attribute
 Add Null check within executeWithSecurity() to avoid crash (#829)
 Add NGSIv2 metadata support to attributeAlias plugin.
+Add mongodb authentication. (#844)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]:
```suggestion
Add mongodb authentication: IOTA_MONGO_USER, IOTA_MONGO_PASSWORD and IOTA_MONGO_AUTH_DB (#844)
```

**Comment by fgalan** [comment]
Thanks for the contribution! It looks nice :) I have provide only some minor feedback, related with documentation.

File: doc/installationguide.md (hunk 2 above)
Comment (SamuelTJackson) [reply doc/installationguide.md]: I renamed authDB to authSource. So it is directly clear what is set in the connection string by the MongoDB.

**Pull request merged**